docs/specs/sections/implementation.tex

```diff
@@ -0,0 +1,89 @@
+\end{flushleft}
+
+The diagram illustrates the following flow of data and interactions between participants:
+
+\begin{itemize}%[label=$\bullet$]
+	\item User submits request to a License Provider by issuing blockchain transaction.
+	\item License provider scans blockchain for requests and obtains the request.
+	\item License provider, upon necessary verification, issues a license.
+	\item License provider sends the license to License Contract via a smart contract call transaction.
+	\item User obtains licenses for a given block-height range.
+	\item User filters out licenses addressed to her/him.
+	\item User calculates a proof (the proof calculation might be delegated).
+	\item User calls \textit{use-license} to redeem a license, via a smart contract call.
+	\item License Contract attempts to verify the proof and, if verified, adds a new session to a list of sessions.
+	\item User requests a service from a Service Provider (off-chain).
+	\item Service Provider asks contract for a session.
+	\item Service Provider grants service to the user (off-chain).
+\end{itemize}
+
+
+\subsection{License Contract}
+
+\begin{flushleft}
+License contract maintains state consisting of the following data:
+\end{flushleft}
+
+\begin{itemize}%[label=$\bullet$]
+	\item List of sessions.
+	\item Map of licenses and their positions in the Merkle tree.
+	\item Merkle tree of license hashes.
+\end{itemize}
+
+
+\begin{flushleft}
+Contract provides the following methods:
+\end{flushleft}
+
+\begin{itemize}%[label=$\bullet$]
+	\item \textit{issue-license}
+	\item \textit{get-licenses}
+	\item \textit{use-license}
+	\item \textit{get-session}
+\end{itemize}
+
+
+\begin{flushleft}
+\textit{issue-license} adds a license to a Merkle tree of licenses. \textit{get-licenses} provides a list of new licenses added in a given block-height range \textit{use-license} attempts to verify the proof and, if verified, adds a new session to a list of sessions, nullifies the license in the Merkle tree. \textit{get-session} finds a session in a list of sessions and returns it to the caller.
```

Comment:
Why not put these in the list above?
Like:
 \item \textit{issue-license}: adds a license to a Merkle tree of licenses

Reply:
* adds a new session to a list of sessions, `and` nullifies the license in the Merkle tree